### PR TITLE
Support the graceful stop for the query node

### PR DESCRIPTION
**Pull request by SimFG**
issue: #20550
origin pr: #20791
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

File: internal/querycoordv2/balance/rowcount_based_balancer.go [hunk 1]
```diff
@@ -177,19 +200,92 @@ outer:
...
 		plans = append(plans, plan)
 		node.setPriority(node.getPriority() + int(s.GetNumOfRows()))
 		queue.push(node)
 	}
-	return plans, nil
+	return plans, b.getChannelPlan(replica, stoppingNodesSegments)
+}
+
+func (b *RowCountBasedBalancer) handleStoppingNodes(replica *meta.Replica, nodeSegments map[int64][]*meta.Segment) ([]SegmentAssignPlan, []ChannelAssignPlan) {
+	segmentPlans := make([]SegmentAssignPlan, 0)
```
Comment (jaime0815) [inline internal/querycoordv2/balance/rowcount_based_balancer.go:227]: suggest using a max length

File: scripts/stop_graceful.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+start=$(date +%s)
+while :
+do
+  sleep 1
+  if [ -z $(get_milvus_process) ]; then
+    echo "Milvus stopped"
+    break
+  fi
+  if [ $(( $(date +%s) - $start )) -lt 30 ]; then
+    echo "Milvus timeout stopped"
```
Comment (jaime0815) [inline scripts/stop_graceful.sh:38]: is should force kill after timeout?

**Pull request merged**